src/knock.ts

```diff
@@ -34,13 +41,20 @@ class Knock {
   readonly messages = new Messages(this);
   readonly tenants = new Tenants(this);
 
-  constructor(readonly key?: string, readonly options: KnockOptions = {}) {
-    if (!key) {
-      this.key = process.env.KNOCK_API_KEY;
+  constructor(keyConfig?: string | KnockKeys, readonly options: KnockOptions = {}) {
+    this.key = (typeof keyConfig === 'string' ? keyConfig : keyConfig?.apiKey) ?? process.env.KNOCK_API_KEY as string;
+    if (!this.key)   throw new NoApiKeyProvidedException();
+    
+    this.signingKey = prepareSigningKey(keyConfig);
```

Comment:
Hmm we really shouldn't need them to instantiate the `Knock` instance with a signing key here given that the node SDK uses the secret key always and the user token is only needed for the public key